src/Compilers/VisualBasic/Portable/Lowering/LocalRewriter/LocalRewriter.vb

```diff
@@ -362,26 +365,25 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             End Get
         End Property
 
-        Private ReadOnly Property GenerateDebugInfo As Boolean
+        Private ReadOnly Property Instrument As Boolean
             Get
-                Return _globalGenerateDebugInfo AndAlso Not _inExpressionLambda
+                Return _instrumenter IsNot Instrumenter.NoOp AndAlso Not _inExpressionLambda
```

Comment:
Unfortunately, there are several places where LocalRewriter still adds **hidden** sequence points. They are not associated with any particular user code and I didn't think it is useful to complicate instrumentation infrastructure with APIs specifically for this purpose. However, we still don't want those sequence points to be added when LocalRewriter is called for lambdas, the condition allows us to not do that. Allowing ```_instrumenter``` to be Nothing is not a bad idea though. Perhaps, it should be Nothing in that scenario. I will open an issue for that.